cms/lib/xblock/tagging.py

```diff
@@ -86,7 +98,46 @@ def student_view_aside(self, block, context):
                     'values': tag.allowed_values,
                     'current_value': self.saved_tags.get(tag.key, None),
                 })
-            return Fragment(render_to_string('structured_tags_block.html', {'tags': tags}))
-            #return Fragment(u'<div class="xblock-render">Hello world!!!</div>')
+            fragment = Fragment(render_to_string('structured_tags_block.html', {'tags': tags}))
+            fragment.add_javascript_url(self._get_studio_resource_url('/js/xblock_asides/structured_tags.js'))
+            fragment.initialize_js('StructuredTagsInit')
+            return fragment
         else:
             return Fragment(u'')
+
+    @XBlock.handler
+    def save_tags(self, request=None, suffix=None):  # pylint: disable=unused-argument
+        """
+        Handler to save choosen tags with connected XBlock
+        """
+        found = False
+        if 'tag' not in request.params:
+            return Response("The required parameter 'tag' is not passed", status=400)
+
+        tag = request.params['tag'].split(':')
+
+        for av_tag in self.available_tags:
+            if av_tag.key == tag[0]:
+                if tag[1] in av_tag.allowed_values:
+                    self.saved_tags[tag[0]] = tag[1]
+                    found = True
+                elif tag[1] == '':
+                    self.saved_tags[tag[0]] = None
+                    found = True
+
+        if not found:
+            return Response("Invalid 'tag' parameter", status=400)
+
+        usage_key = None
+        for path_part in request.path.split('/'):
+            if path_part.startswith(AsideUsageKeyV1.CANONICAL_NAMESPACE):
+                usage_key = UsageKey.from_string(path_part)
+                break
+
+        if usage_key:
+            descriptor = modulestore().get_item(usage_key.usage_key)
+            modulestore().update_item(descriptor, request._request.user.id, asides=[self])  # pylint: disable=protected-access
```

Comment:
This is breaking an abstraction barrier. XBlocks shouldn't be calling Modulestore code. Better would be to have the code that is calling the handler on the aside call update_item and pass in the (possibly modified) aside.